esrally/racecontrol.py

```diff
@@ -195,9 +195,12 @@ def setup(self, msg, sender):
         # cases (external pipeline and user did not specify the distribution version) where we need to derive it ourselves. For source
         # builds we always assume "master"
         if not msg.sources and not self.cfg.exists("mechanic", "distribution.version"):
-            distribution_version = mechanic.cluster_distribution_version(self.cfg)
-            if not distribution_version:
-                raise exceptions.SystemSetupError("A distribution version is required. Please specify it with --distribution-version.")
+            import elasticsearch
+            try:
+                distribution_version = mechanic.cluster_distribution_version(self.cfg)
+            except elasticsearch.ConnectionError as e:
```

Comment:
Is there a reason we are only catching connection errors? Does it make sense to catch the base type `elasticsearch.ElasticsearchException` instead? Also, we are leaking implementation details (usage of the Elasticsearch Python client) and I wonder whether we should push down that exception handling code into the callee `mechanic#cluster_distribution_version()`?

Reply:
Pushing the exception handling down brings us back to the situation we had before this change.
I think catching anything besides ConnectionError here risks another misleading error message.  I've decided "friendliness" isn't worth it here and reverted to the "unfriendly" version.

Reply:
> Pushing the exception handling down brings us back to the situation we had before this change.

Not really: Previously we were catching any exception and returning `None`. Now we are catching an exception and reraise a friendlier one so this is definitely a change in behavior.  Anyway, I'm ok with your change in https://github.com/elastic/rally/pull/814/commits/61889c6c90938187371b9e50c2c728171450a930 so I'll approve.